File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/HeapRepository.java
package com.github.ol_loginov.heaplibweb.repository.heap;


import com.github.ol_loginov.heaplibweb.repository.HeapFile;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface HeapRepository extends CrudRepository<HeapEntity, Integer> {
	Optional<HeapEntity> findOneByFile(HeapFile heapFile);

	List<HeapEntity> findAllByOrderByIdDesc();
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/HeapRepository.kt
package com.github.ol_loginov.heaplibweb.repository.heap

interface HeapRepository {
    fun persist(entity: HeapEntity): HeapEntity
    fun merge(entity: HeapEntity)

    fun findById(id: Int): HeapEntity?
    fun findByFileLatest(path: String): HeapEntity?
    fun findAllOrderByIdDesc(): List<HeapEntity>
    fun getScope(heapEntity: HeapEntity): HeapScope
}